src/containerapp/azext_containerapp/_cloud_build_utils.py

```diff
@@ -130,27 +130,32 @@ def spin():
 
         # Source code compression
         done_spinner = False
-        thread = display_spinner(f"Compressing data: {font_bold}{source}{font_default}")
-        tar_file_path = os.path.join(tempfile.gettempdir(), f"{build_name}.tar.gz")
-        archive_source_code(tar_file_path, source)
+        source_file_path = source
+        if os.path.isdir(source):
+            thread = display_spinner(f"Compressing data: {font_bold}{source}{font_default}")
+            source_file_path = os.path.join(tempfile.gettempdir(), f"{build_name}.tar.gz")
+            archive_source_code(source_file_path, source)
         done_spinner = True
         thread.join()
```

Comment:
Could you please move the `done_spinner = False` and the equivalent lines in the `if` block?
```
# Source code compression (only compressed if not an artifact already)
        source_file_path = source
        if os.path.isdir(source):
            done_spinner = False
            thread = display_spinner(f"Compressing data: {font_bold}{source}{font_default}")
            source_file_path = os.path.join(tempfile.gettempdir(), f"{build_name}.tar.gz")
            archive_source_code(source_file_path, source)
            done_spinner = True
            thread.join()
```